osquery/core/http.cpp

```diff
@@ -0,0 +1,97 @@
+/*
+ *  Copyright (c) 2014, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+
+#include <osquery/core.h>
+#include <osquery/database.h>
+#include <osquery/logger.h>
+#include "osquery/core/http.h"
+
+#include <curl/curl.h>
+
+namespace osquery{
+
+	std::string decodeError(HTTPResponse resp){
+		long code = resp.getReturnCode();
+
+		switch(code){
+			case -1: 	return "cURL could not initalize.";
+			case -2: 	return "Could not parse the JSON";
+			case 200:	return "Success";
+			case 204:	return "No conent";
+			case 301:	return "Moved permanently";
+			case 304:	return "Not modified";
+			case 400:	return "Bad request";
+			case 401:	return "Unauthorized";
+			case 403:	return "Forbidden";
+			case 404:	return "Page not found";
+			case 405:	return "Method not allowed";
+			case 406:	return "Unacceptable";
+			case 411:	return "Length required";
+			default:	return "Unknown code";
+		}
+	}
+
+	HTTPResponse::HTTPResponse(long code){
+		return_code = code;
+	}
+
+	HTTPResponse::HTTPResponse(long code, std::string contents){
+		return_code = code;
+		this->contents = contents;
+	}
+
+	long HTTPResponse::getReturnCode(){
+		return return_code;
+	}
+
+	std::string HTTPResponse::getRaw(){
+		return contents;
+	}
+
+	int HTTPResponse::getJSON(pt::ptree& tree){
```

Comment:
Can you better describe this interface by declaring the methods which do not alter state as const?

```cpp
int HTTPResponse::getJSON(pt::ptree& tree) const {
```